home/import_assessment_pages.py

```diff
@@ -0,0 +1,361 @@
+# do a single write to the database from, and encapsulate all that complexity
+@dataclass(slots=True)
+class ShadowAssessmentPage:
+    slug: str
+    locale: Locale
+    row_num: int
+    title: str = ""
+    tags: list[str] = field(default_factory=list)
+    high_result_page: list[str] = field(default_factory=list)
+    medium_result_page: list[str] = field(default_factory=list)
+    low_result_page: list[str] = field(default_factory=list)
+    high_inflection: str = ""
+    medium_inflection: str = ""
+    low_inflection: str = ""
+    generic_error: str = ""
+    questions: list[str] | list[dict[str, Any]] | str = field(default_factory=list)
+    error: str = ""
+    answers: list[str] = field(default_factory=list)
+    scores: list[str] = field(default_factory=list)
+
+    def save(self, parent: Page) -> None:
+        try:
+            page = Assessment.objects.get(slug=self.slug, locale=self.locale)
+        except Assessment.DoesNotExist:
+            page = Assessment(slug=self.slug, locale=self.locale)
+        self.add_identifiers_to_page(page)
+        self.add_tags_to_page(page)
+        try:
+            with contextlib.suppress(NodeAlreadySaved):
+                parent.add_child(instance=page)
+            page.save_revision().publish()
+        except ValidationError as err:
+            raise ImportAssessmentException(f"Validation error: {err}", self.row_num)
+
+    def add_identifiers_to_page(self, page: Assessment) -> None:
+        page.slug = self.slug
+        page.title = self.title
+        page.locale = self.locale
+        page.high_inflection = self.high_inflection
+        page.high_result_page_id = result_page_get_id_from_slug(self.high_result_page)
+        page.medium_inflection = self.medium_inflection
+        page.medium_result_page_id = result_page_get_id_from_slug(
+            self.medium_result_page
+        )
+        page.low_inflection = self.low_inflection
+        page.low_result_page_id = result_page_get_id_from_slug(self.low_result_page)
+        page.generic_error = self.generic_error
+        page.questions = create_question_streamfield(self.questions)
+
+    def add_tags_to_page(self, page: Assessment) -> None:
+        page.tags.clear()
+        for tag_name in self.tags:
+            tag, _ = Tag.objects.get_or_create(name=tag_name)
+            page.tags.add(tag)
+
+
+@dataclass(slots=True, frozen=True)
+class ContentRow:
+    slug: str
+    title: str = ""
+    page_id: int | None = None
+    tags: list[str] = field(default_factory=list)
+    locale: str = ""
+    high_result_page: list[str] = field(default_factory=list)
+    high_inflection: str = ""
+    medium_result_page: list[str] = field(default_factory=list)
+    medium_inflection: str = ""
+    low_result_page: list[str] = field(default_factory=list)
+    low_inflection: str = ""
+    generic_error: str = ""
+    question_count: str = ""
+    question: list[str] | list[dict[str, Any]] | str = field(default_factory=list)
```

Comment:
Why would this be a list of strings? Surely every row only contains a single question?